plugins/jobs/src/js/components/JobStopRunModal.js

```diff
@@ -0,0 +1,86 @@
+import { Confirm } from "reactjs-components";
+import PropTypes from "prop-types";
+import React from "react";
+import ModalHeading from "#SRC/js/components/modals/ModalHeading";
+
+class JobStopRunModal extends React.Component {
+  getContentHeader(selectedItems, selectedItemsLength) {
+    let headerContent = ` ${selectedItemsLength} Job Runs`;
+    if (selectedItemsLength === 1) {
+      headerContent = "this";
+    }
+
+    return (
+      <ModalHeading key="confirmHeader">
+        {`Are you sure you want to stop ${headerContent}?`}
+      </ModalHeading>
+    );
+  }
+
+  getConfirmTextBody(selectedItems, selectedItemsLength) {
+    let bodyText;
+
+    if (selectedItemsLength === 1) {
+      bodyText = `the job run with id ${selectedItems[0]}`;
+    } else {
+      bodyText = "the selected job runs";
+    }
+
+    return <span key="confirmText">You are about to stop {bodyText}.</span>;
+  }
+
+  getModalContents() {
+    const { selectedItems } = this.props;
+    const selectedItemsLength = selectedItems.length;
+
+    return (
+      <div className="text-align-center">
+        {this.getConfirmTextBody(selectedItems, selectedItemsLength)}
+      </div>
+    );
+  }
+
+  render() {
+    const { onClose, open, selectedItems } = this.props;
+    let rightButtonText = "Stop Job Run";
+    const selectedItemsLength = selectedItems.length;
```

Comment:
We assign `selectedItems.length` and then use it again 1 line after.

Reply:
There is a lot of things I don't like on this component, but it a plain UI component from which I took of all the data logic to do the data layer refactoring. I don't plan to rewrite this.

Reply:
Daniel made a comment here https://github.com/dcos/dcos-ui/pull/3043/files, that made me inline some code that made me fix that issue on #3043.